File: CSharp/leetcodesln/RemoveDuplicatesFromSortedList.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace leetcodesln
{
    public class RemoveDuplicatesFromSortedList
    {
        public ListNode DeleteDuplicates(ListNode head)
        {
            if (head == null || head.next == null) return head;

            head.next = DeleteDuplicates(head.next);
            return head = head.val == head.next.val ? head.next : head;
        }
    }
}

File: Go/questions/RemoveDuplicatesFromSortedList.go
package questions

import "github.com/leetcode/common"

func DeleteDuplicates(head *common.ListNode) *common.ListNode {
	if head == nil || head.Next == nil {
		return head
	}

	head.Next = DeleteDuplicates(head.Next)
	if head.Val == head.Next.Val {
		head = head.Next
	}
	return head
}
